models/cm1/model_mod.f90

```diff
@@ -1127,6 +1129,37 @@ function observation_on_grid(obs_location, ndims)
       return ! exit early
 
    endif
+elseif ( periodic_x) then
+    ! require that the point is contained within the staggered grid for the 
+    ! y - direction since you cannot wrap-arround
+    if ( (obs_location(1) < xf(1)) .or. (obs_location(1) > xf(nip1)) .or. &
+        (obs_location(2) < yh(1)) .or. (obs_location(2) > yh(njp1)) ) then
```

Comment:
The yh grid is nj long, so go off the end of the array with njp1.

```suggestion
    if ( (obs_location(1) < xf(1)) .or. (obs_location(1) > xf(nip1)) .or. &
        (obs_location(2) < yh(1)) .or. (obs_location(2) > yh(nj)) ) then
```

Maybe I am mis-understanding the comment, but the code seems opposite to the comment here:

non-staggered x would be xh
staggered in y would be yf

Reply:
You are correct, the updates look good.